### DISCUSSION TITLE
In-game multiplayer chat recieves input from left and right buttons whilst playing

### DISCUSSION THREAD

**Discussion by brynblack** [ppy/osu#16125] [Q&A]
**Description:**
When playing a multiplayer match and you open the in-game chat and resume playing the beatmap, the chat continues to stay open and receive input from left and right keyboard buttons (in my case, these were Q and E). This seems unintentional because I doubt anyone would be typing messages whilst clicking circles, and frankly it gets quite irritating, especially when you accidentally hit your TAB key mid-game and start hearing typing sounds.

**Example:**

https://user-images.githubusercontent.com/49110391/146495556-f661a6fd-54a3-41b7-a7fb-b2832559b58b.mp4

**Logs:**
[database.log](https://github.com/ppy/osu/files/7732205/database.log)
[network.log](https://github.com/ppy/osu/files/7732216/network.log)
[performance.log](https://github.com/ppy/osu/files/7732207/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7732208/runtime.log)
[updater.log](https://github.com/ppy/osu/files/7732209/updater.log)

**Specs:**
Ryzen 5 3600
RTX 2070 Super
G.Skill Trident Z RGB 16GB
Windows 10 Pro
osu! 2021.1211.0-lazer

**Comment by peppy** [comment]
Weird, it should have automatically defocused. Either that has regressed or something weird about your setup has meant that it didn't happen.

**Reply by brynblack** [reply]
I pressed TAB at the start of the beatmap to quickly demonstrate the issue, rather than playing a bit into the beatmap and then pressing TAB, which is why it didn't automatically defocus upon loading. Sorry if it caused any confusion, I should've pointed that out more clearly. I've also confirmed that this issue is not due to my setup, as I asked a friend to try it and they were able to reproduce it on the same version of lazer.

**Reply by peppy** [reply]
Oh, I see.

Pressing tab *intentionally* does focus the chat. Not sure whether this should be changed.

**Reply by brynblack** [reply]
But the issue is that the chat was receiving input from the left and right keyboard buttons while playing the beatmap, so the chat just filled up with "qeqeqeqeqeqeqe..." but you could still click circles at the same time, and the sound effect for typing could be heard while clicking circles, which doesn't seem intended.

**Reply by bdach** [reply]
The thing is that pressing tab when gameplay is ongoing expresses intent to type. The action probably shouldn't happen in both places, so maybe the textbox should block the gameplay actions? Or the other way around, ruleset bindings should not be typeable mid gameplay? Both seem like suboptimal choices in some scenarios.

**Reply by peppy** [reply]
I'd say blocking gameplay is the best option, if we are going with blocking anything. Not sure it needs to change though.

**Reply by Theighlin** [reply]
A bit different, but i thought i might as well mention it here given the topic of the discussion. I've seen multiple first-time lazer players getting confused for trying to toggle hud via alt-tab and finding themselves typing instead. This is because chat takes focus as soon as you press tab, so if you press them in the tab-alt order, the hud shortcut won't work. It' likely a one off thing for them that they won't repeat, but chat should probably take focus after releasing the tab key.
